Handle deprecation and removal of ``transformClient``

See: https://github.com/robotology/yarp/pull/2999

Related issues: 
- https://github.com/robotology/walking-teleoperation/issues/131
- https://github.com/robotology/wearables/issues/191
- https://github.com/ami-iit/yarp-openvr-trackers/issues/13

I started checking how to handle this, but I immediately found an issue. If I try to launch
```
yarprobotinterface --config fts_yarp_only.xml
```
I get the following error
```
[ERROR] |yarp.os.Property| cannot read from yarprobotinterface.ini
[DEBUG] Reading file 
[ERROR] Syntax error while loading  at line 0 . Failed to open file
[FATAL] Config file    not parsed correctly.
Trace requested at /home/sdafarra/Software/robotology-superbuild/src/YARP/src/yarprobotinterface/Module.cpp:132 by code called from:
/home/sdafarra/Software/robotology-superbuild/build/install/lib/libYARP_os.so.3(_Z16yarp_print_traceP8_IO_FILEPKcj+0x3b) [0x7f99dad78f2b]
yarprobotinterface(+0x9038) [0x55bd3e833038]
yarprobotinterface(+0xa439) [0x55bd3e834439]
/home/sdafarra/Software/robotology-superbuild/build/install/lib/libYARP_os.so.3(_ZN4yarp2os8RFModule9runModuleERNS0_14ResourceFinderE+0x77) [0x7f99dadd4647]
yarprobotinterface(+0x8b90) [0x55bd3e832b90]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf3) [0x7f99da8cf083]
yarprobotinterface(+0x8c5e) [0x55bd3e832c5e]
```
Indeed, it seems that the file ``fts_yarp_only.xml`` is not installed.

--- a/src/devices/openxrheadset/OpenXrHeadset.cpp
+++ b/src/devices/openxrheadset/OpenXrHeadset.cpp
@@ -39,6 +39,16 @@ bool yarp::dev::OpenXrHeadset::open(yarp::os::Searchable &cfg)
         m_prefix = name;
     }
 
+    std::string rpcName = cfg.check("rpc_name", yarp::os::Value("rpc")).toString();
+    if (rpcName.front() != '/')
+    {
+        m_rpcPortName = m_prefix +  '/' + rpcName;
+    }
+    else
+    {
+        m_rpcPortName = m_prefix + rpcName;
+    }
+
     //checking the additional guis parameter in the configuration file..
     {
         constexpr unsigned int STRING = 0;
@@ -442,9 +452,9 @@ bool yarp::dev::OpenXrHeadset::threadInit()
         std::lock_guard<std::mutex> lock(m_mutex);
 
         this->yarp().attachAsServer(this->m_rpcPort);
-        if(!m_rpcPort.open(m_prefix + "/rpc"))
+        if(!m_rpcPort.open(m_rpcPortName))
         {
-            yCError(OPENXRHEADSET) << "Could not open" << m_prefix + "/rpc" << " RPC port.";
+            yCError(OPENXRHEADSET) << "Could not open" << m_rpcPortName << " RPC port.";
             return false;
         }
     }

--- a/src/devices/openxrheadset/OpenXrHeadset.cpp
+++ b/src/devices/openxrheadset/OpenXrHeadset.cpp
@@ -307,6 +307,13 @@ bool yarp::dev::OpenXrHeadset::open(yarp::os::Searchable &cfg)
     yarp::os::Property tfClientCfg;
     tfClientCfg.put("device", cfg.check("tfDevice", yarp::os::Value("frameTransformClient")).asString());
     tfClientCfg.put("filexml_option",  cfg.check("tfFile", yarp::os::Value("ftc_yarp_only.xml")).asString());
+    tfClientCfg.put("ft_client_prefix", cfg.check("tfLocal", yarp::os::Value(m_prefix + "/tf")).asString());
+    if (cfg.check("tfRemote"))
+    {
+        tfClientCfg.put("ft_server_prefix", cfg.find("tfRemote").asString());
+    }
+    tfClientCfg.put("period", period);
+    tfClientCfg.put("local_rpc", m_prefix + "/tf/local_rpc");
 
     if (!m_driver.open(tfClientCfg))
     {

--- a/src/devices/openxrheadset/OpenXrHeadset.cpp
+++ b/src/devices/openxrheadset/OpenXrHeadset.cpp
@@ -305,9 +305,8 @@ bool yarp::dev::OpenXrHeadset::open(yarp::os::Searchable &cfg)
 
     //opening tf client
     yarp::os::Property tfClientCfg;
-    tfClientCfg.put("device", cfg.check("tfDevice", yarp::os::Value("transformClient")).asString());
-    tfClientCfg.put("local",  cfg.check("tfLocal", yarp::os::Value(m_prefix + "/tf")).asString());
-    tfClientCfg.put("remote", cfg.check("tfRemote", yarp::os::Value("/transformServer")).asString());
+    tfClientCfg.put("device", cfg.check("tfDevice", yarp::os::Value("frameTransformClient")).asString());
+    tfClientCfg.put("filexml_option",  cfg.check("tfFile", yarp::os::Value("ftc_yarp_only.xml")).asString());
 
     if (!m_driver.open(tfClientCfg))
     {
@@ -320,7 +319,7 @@ bool yarp::dev::OpenXrHeadset::open(yarp::os::Searchable &cfg)
         yCError(OPENXRHEADSET) << "Unable to view IFrameTransform interface.";
         return false;
     }
-    yCInfo(OPENXRHEADSET) << "TransformCLient successfully opened at port: " << cfg.find("tfLocal").asString();
+    yCInfo(OPENXRHEADSET) << "Transform client successfully opened.";
 
     FilteredPosePublisherSettings posePublisherSettings;
     posePublisherSettings.tfPublisher = m_tfPublisher;

--- a/src/devices/openxrheadset/OpenXrHeadset.h
+++ b/src/devices/openxrheadset/OpenXrHeadset.h
@@ -310,6 +310,7 @@ private:
 
     OpenXrInterface::Pose m_rootFrameRawHRootFrame;
 
+    std::string m_rpcPortName;
     yarp::os::Port m_rpcPort;
 
     std::atomic_bool m_closed{ false };

--- a/src/utils/OpenXrFrameViz/main.cpp
+++ b/src/utils/OpenXrFrameViz/main.cpp
@@ -98,6 +98,12 @@ int main(int argc, char** argv)
     yarp::os::Property tfClientCfg;
     tfClientCfg.put("device", rf.check("tfDevice", yarp::os::Value("frameTransformClient")).asString());
     tfClientCfg.put("filexml_option",  rf.check("tfFile", yarp::os::Value("ftc_yarp_only.xml")).asString());
+    tfClientCfg.put("ft_client_prefix", rf.check("tfPrefix", yarp::os::Value("/OpenXrFrameViewer/tf")).asString());
+    if (rf.check("tfRemote"))
+    {
+        tfClientCfg.put("ft_server_prefix", rf.find("tfRemote").asString());
+    }
+    tfClientCfg.put("local_rpc", "OpenXrFrameViewer/tf/local_rpc");
 
     yarp::dev::PolyDriver driver;
     yarp::dev::IFrameTransform* tfReader;

--- a/src/utils/OpenXrFrameViz/main.cpp
+++ b/src/utils/OpenXrFrameViz/main.cpp
@@ -95,15 +95,17 @@ int main(int argc, char** argv)
         return EXIT_FAILURE;
     }
 
+    std::string name = rf.check("name", yarp::os::Value("OpenXrFrameViewer")).asString();
+
     yarp::os::Property tfClientCfg;
     tfClientCfg.put("device", rf.check("tfDevice", yarp::os::Value("frameTransformClient")).asString());
     tfClientCfg.put("filexml_option",  rf.check("tfFile", yarp::os::Value("ftc_yarp_only.xml")).asString());
-    tfClientCfg.put("ft_client_prefix", rf.check("tfPrefix", yarp::os::Value("/OpenXrFrameViewer/tf")).asString());
+    tfClientCfg.put("ft_client_prefix", "/" + name + "/tf");
     if (rf.check("tfRemote"))
     {
         tfClientCfg.put("ft_server_prefix", rf.find("tfRemote").asString());
     }
-    tfClientCfg.put("local_rpc", "OpenXrFrameViewer/tf/local_rpc");
+    tfClientCfg.put("local_rpc", "/" + name + "/tf/local_rpc");
 
     yarp::dev::PolyDriver driver;
     yarp::dev::IFrameTransform* tfReader;

--- a/src/utils/OpenXrFrameViz/main.cpp
+++ b/src/utils/OpenXrFrameViz/main.cpp
@@ -96,9 +96,8 @@ int main(int argc, char** argv)
     }
 
     yarp::os::Property tfClientCfg;
-    tfClientCfg.put("device", "transformClient");
-    tfClientCfg.put("local",  rf.check("tfLocal", yarp::os::Value("/OpenXrFrameViewer/tf")).asString());
-    tfClientCfg.put("remote", rf.check("tfRemote", yarp::os::Value("/transformServer")).asString());
+    tfClientCfg.put("device", rf.check("tfDevice", yarp::os::Value("frameTransformClient")).asString());
+    tfClientCfg.put("filexml_option",  rf.check("tfFile", yarp::os::Value("ftc_yarp_only.xml")).asString());
 
     yarp::dev::PolyDriver driver;
     yarp::dev::IFrameTransform* tfReader;

